pageserver/src/consumption_metrics.rs

```diff
@@ -374,45 +378,31 @@ async fn collect_metrics_iteration(
             },
         }));
 
-        const MAX_RETRIES: u32 = 3;
-
-        for attempt in 0..MAX_RETRIES {
-            let res = client
-                .post(metric_collection_endpoint.clone())
-                .json(&EventChunk {
-                    events: (&chunk_to_send).into(),
-                })
-                .send()
-                .await;
-
-            match res {
-                Ok(res) => {
-                    if res.status().is_success() {
-                        // update cached metrics after they were sent successfully
-                        for (curr_key, curr_val) in chunk.iter() {
-                            cached_metrics.insert(curr_key.clone(), *curr_val);
-                        }
-                    } else {
-                        error!("metrics endpoint refused the sent metrics: {:?}", res);
-                        for metric in chunk_to_send
-                            .iter()
-                            .filter(|metric| metric.value > (1u64 << 40))
-                        {
-                            // Report if the metric value is suspiciously large
-                            error!("potentially abnormal metric value: {:?}", metric);
-                        }
-                    }
-                    break;
-                }
-                Err(err) if err.is_timeout() => {
-                    error!(attempt, "timeout sending metrics, retrying immediately");
-                    continue;
-                }
-                Err(err) => {
-                    error!(attempt, ?err, "failed to send metrics");
-                    break;
-                }
+        use bytes::BufMut;
+
+        // FIXME: this is a new panic we did not have before. not really essential, but panics from
+        // this task currently restart pageserver.
+        serde_json::to_writer(
+            (&mut buffer).writer(),
+            &EventChunk {
+                events: (&chunk_to_send).into(),
+            },
+        )
+        .expect("serialization must not fail");
+
+        let body = buffer.split().freeze();
+
+        let res = upload(client, metric_collection_endpoint, body, &cancel).await;
+
+        if res.is_ok() {
+            for (curr_key, curr_val) in chunk.iter() {
+                cached_metrics.insert(curr_key.clone(), *curr_val);
             }
+        } else {
+            // no need to log, backoff::retry has done it, just give up uploading, but still finish
+            // up the on-disk part.
+            //
+            // note if we are cancelled we'll end up just writing the files to disk asap
```

Comment:
this comment is also about future work...

Reply:
Dealt in [a8a0ec1](https://github.com/neondatabase/neon/pull/5317/commits/a8a0ec1e8ddf6f461ecad438550cf411ccb75b4c)